aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1465,4 +1465,30 @@ Tensor& diag_out(Tensor &result, const Tensor& self, int64_t dimension) {
   return result;
 }
 
+Tensor trace(const Tensor& self) {
+  Tensor result = at::empty({}, self.options());
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "trace", [&] {
+    scalar_t sum = 0;
+    auto t_data = self.data_ptr<scalar_t>();
```

Comment:
nit: Use "const auto*" to indicate it's a pointer and that we're not modifying it

Reply:
done